### PR TITLE
serializers

### PR DESCRIPTION
**Pull request by omarelsheekh**

### CODE REVIEW

**Comment by abom** [comment]
I don't like having a single interface for all serializers, also, i need to access yaml for example as

`j.serializers.yaml.load`, not the way it's implemented now.

File: jumpscale/data/serializers/base64.py [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+
+
+def encode(s):
+    if isinstance(s,str):
+        b = s.encode()
+    else:
+        b = s
+    return base64.b64encode(b)
+
+def dumps(s):
```
Comment (abom) [inline jumpscale/data/serializers/base64.py]: We don't need dumps/loads here, encode/decode is enough.

File: jumpscale/data/serializers/blosc.py [hunk 2]
```diff
@@ -0,0 +1,9 @@
+from blosc import compress, decompress
+
+
+def dumps(obj):
```
Comment (abom) [inline jumpscale/data/serializers/blosc.py]: See previous comment, do not follow the same naming for all serializes, here compress/decompress is the right one, you can just do as following:
```
import blosc

def compress(input):
    # first...convert input to byte-array if possible
   return blosc.comporess(input)
```

File: jumpscale/data/serializers/lzma.py [hunk 3]
```diff
@@ -0,0 +1,7 @@
+import pylzma
+
+def dumps(self, obj):
```
Comment (abom) [inline jumpscale/data/serializers/lzma.py]: The same here

File: jumpscale/data/serializers/toml.py [hunk 4]
```diff
@@ -0,0 +1,111 @@
+import pytoml
+def fancydumps(self, obj, secure=False):
```
Comment (abom) [inline jumpscale/data/serializers/toml.py]: Please remove this fancy code that won't work.

**Comment by abom** [comment]
I see some parts where you need to encode/decode strings and byte arrays, you simply can add some utilities/helpers to do so, and use them inside all serializers.

File: jumpscale/data/serializers/base64.py (hunk 1 above)
Comment (omarelsheekh) [reply jumpscale/data/serializers/base64.py]: i did it as old code

File: jumpscale/data/serializers/blosc.py (hunk 2 above)
Comment (omarelsheekh) [reply jumpscale/data/serializers/blosc.py]: in my code i gave you options to use compress,decompress,dumps and loads methods

File: jumpscale/data/serializers/toml.py (hunk 4 above)
Comment (omarelsheekh) [reply jumpscale/data/serializers/toml.py]: solved in the last commit

File: jumpscale/data/serializers/lzma.py (hunk 3 above)
Comment (omarelsheekh) [reply jumpscale/data/serializers/lzma.py]: solved in the last commit

File: jumpscale/data/serializers/base64.py (hunk 1 above)
Comment (abom) [reply jumpscale/data/serializers/base64.py]: IMHO, old code has bad naming.

File: jumpscale/data/serializers/blosc.py (hunk 2 above)
Comment (abom) [reply jumpscale/data/serializers/blosc.py]: As the previous comment, dumps/loads is not the right naming.

**Comment by abom** [comment]
Did you see this comment?

https://github.com/js-next/js-ng/pull/28#issuecomment-518732799

**Comment by omarelsheekh** [comment]
string.encode  only repeated here 2 times , no need for helper

**Pull request merged**